roles/openshift_cfme/tasks/main.yml

```diff
@@ -0,0 +1,176 @@
+# PASSED IN ROLE VARIABLES:
+#
+# - template_dir - A remote temp dir to template the PVs in
+
+---
+- name: What are the groups?
+  debug:
+    var: groups.keys()
+
+- name: Where is my template dir
+  debug:
+    var: template_dir
+
+######################################################################
+- name: Ensure the CFME user exists
+  oc_user:
+    state: present
+    username: "{{ openshift_cfme_user }}"
+
+- name: Ensure the CFME namespace exists with CFME user as admin
+  oc_project:
+    state: present
+    name: "{{ openshift_cfme_project }}"
+    display_name: "{{ openshift_cfme_project_description }}"
+    admin: "{{ openshift_cfme_user }}"
+
+- name: Ensure the CFME namespace service account is privileged
+  oc_adm_policy_user:
+    namespace: "{{ openshift_cfme_project }}"
+    user: "{{ openshift_cfme_service_account }}"
+    resource_kind: scc
+    resource_name: privileged
+    state: present
+
+######################################################################
+- name: Ensure bulk image import limit is tuned
+  yedit:
+    src: /etc/origin/master/master-config.yaml
+    key: 'imagePolicyConfig#maxImagesBulkImportedPerRepository'
+    value: "{{ openshift_cfme_maxImagesBulkImportedPerRepository | int() }}"
+    state: present
+    backup: True
+  register: master_config_updated
+
+# TODO: Figure out how to link to the right handlers rather than
+# duplicate more restart logic
+#  notify: 'config change - restart master service'
+- name: TODO Ensure master service is restarted if the config changed
+  debug:
+    msg: "Not doing anything yet, just a placeholder"
+  when: master_config_updated.changed
+
+######################################################################
+- name: Create NFS exports for CFME PVs
+  copy:
+    src: openshift_cfme.exports
+    dest: /etc/exports.d/openshift_cfme.exports
+  register: nfs_exports_updated
+
+- name: Ensure the NFS export table is refreshed if exports were added
+  command: exportfs -ar
+  when: nfs_exports_updated.changed
+
+######################################################################
+# This needs to happen on the first master
+- name: Ensure the CFME PV templates are evaluated
+  template:
+    src: "{{ item }}.yaml.j2"
+    dest: "{{ template_dir }}/{{ item }}.yaml"
+  with_items: "{{ openshift_cfme_pv_templates }}"
+
+# BEGIN LONG TODO: There has to be a better way to do this. Must refactor.
```

Comment:
This `LONG TODO` basically says "I'm repeating myself here a lot. There must be a better pattern for doing this"